org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/StringTemplateExpression.java

```diff
@@ -0,0 +1,229 @@
+		addProperty(TEMPLATE_PROCESSOR, propertyList);
+		addProperty(STRING_TEMPLATE_PROPERTY, propertyList);
+		PROPERTY_DESCRIPTORS = reapPropertyList(propertyList);
+	}
+
+	/**
+	 * The pattern type;
+	 */
+	private Expression processor = null;
+
+	/**
+	 */
+	private StringTemplate template = null;
+
+	StringTemplateExpression(AST ast) {
+		super(ast);
+		supportedOnlyIn21();
+	}
+
+	/**
+	 * @noreference This method is not intended to be referenced by clients as it is a part of Java preview feature.
+	 */
+	public void setProcessor(Expression processor) {
+		supportedOnlyIn21();
+		if (processor == null) {
+			throw new IllegalArgumentException();
+		}
+		ASTNode oldChild = this.processor;
+		preReplaceChild(oldChild, processor, TEMPLATE_PROCESSOR);
+		this.processor = processor;
+		postReplaceChild(oldChild, processor, TEMPLATE_PROCESSOR);
+	}
+
+	/**
+	 * @noreference This method is not intended to be referenced by clients as it is a part of Java preview feature.
+	 */
+	public Expression getProcessor() {
+		supportedOnlyIn21();
+		if (this.processor  == null) {
+			// lazy init must be thread-safe for readers
+			synchronized (this) {
+				if (this.processor == null) {
+					preLazyInit();
+					this.processor = new SimpleName(this.ast);
+					postLazyInit(this.processor, TEMPLATE_PROCESSOR);
+				}
+			}
+		}
+		return this.processor;
+	}
+	/**
+	 * @noreference This method is not intended to be referenced by clients as it is a part of Java preview feature.
+	 */
+	public void setTemplate(StringTemplate template) {
+		supportedOnlyIn21();
+		if (template == null) {
+			throw new IllegalArgumentException();
+		}
+		ASTNode oldChild = this.template;
+		preReplaceChild(oldChild, template, STRING_TEMPLATE_PROPERTY);
+		this.template = template;
+		postReplaceChild(oldChild, template, STRING_TEMPLATE_PROPERTY);
+	}
+
+	/**
+	 * @noreference This method is not intended to be referenced by clients as it is a part of Java preview feature.
+	 */
+	public Expression getTemplate() {
+		supportedOnlyIn21();
+		if (this.template  == null) {
+			// lazy init must be thread-safe for readers
+			synchronized (this) {
+				if (this.template == null) {
+					preLazyInit();
+					this.template = new StringTemplate(this.ast);
+					postLazyInit(this.template, TEMPLATE_PROCESSOR);
+				}
+			}
+		}
+		return this.template;
+	}
+	/**
+	 * Returns a list of structural property descriptors for this node type.
+	 * Clients must not modify the result.
+	 *
+	 * @param apiLevel the API level; one of the
+	 * <code>AST.JLS*</code> constants
+
+	 * @return a list of property descriptors (element type:
+	 * {@link StructuralPropertyDescriptor})
+	 * @since 3.24
+	 */
+	public static List propertyDescriptors(int apiLevel) {
+		return PROPERTY_DESCRIPTORS;
+	}
+
+	public static List propertyDescriptors(int apiLevel, boolean previewEnabled) {
+		if (DOMASTUtil.isStringTemplateSupported(apiLevel, previewEnabled)) {
+			return PROPERTY_DESCRIPTORS;
+		}
+		return null;
+	}
```

Comment:
Is it reasonable that only one checks for isStringTemplateSupported ??

Reply:
As discussed offline, the other one will return null anyway as this is preview only. When we promote this as standard, we should fix them all.